src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -425,6 +420,16 @@ public override IReadOnlyList<Endpoint> Endpoints
             }
         }
 
+        private void UpdateEndpoints()
+        {
+            lock (_lock)
```

Comment:
Hrm, I added some double check locking to avoid initializing the endpoints multiple times but I don't like this